lib/charms/alertmanager_k8s/v0/alertmanager_remote_configuration.py

```diff
@@ -0,0 +1,1890 @@
+            config_file_path: The path to the Alertmanager configuration file. Defaults
+                to `/etc/alertmanager/alertmanager.yml`.
+            api_address: Defaults to `http://localhost:9093`
+        """
+        super().__init__(charm, relation_name)
+        self._charm = charm
+        self._relation_name = relation_name
+        self._config_file_path = config_file_path
+        self.api_address = api_address
+
+        on_relation = self._charm.on[self._relation_name]
+
+        self.framework.observe(on_relation.relation_joined, self._on_relation_joined)
+
+    def _on_relation_joined(self, event: RelationJoinedEvent) -> None:
+        """Event handler for RelationJoinedEvent.
+
+        Args:
+            event: Juju RelationJoinedEvent
+
+        Returns:
+            None
+        """
+        self.load_initial_alertmanager_config_to_relation_data_bag(event)
+
+    def load_initial_alertmanager_config_to_relation_data_bag(
+        self, event: RelationJoinedEvent
+    ) -> None:
+        """Loads current Alertmanager configuration to the relation data bag.
+
+        Reads current Alertmanager configuration from alertmanager.yml and puts the content
+        inside relation data bag. If needed, it can be used by `alertmanager-remote-configuration`
+        consumer as a starting point for generating custom configuration.
+
+        Args:
+            event: Juju RelationJoinedEvent
+
+        Returns:
+            None
+        """
+        if not self._charm.unit.is_leader():
+            return
+        alertmanager_config = self._get_current_config()
+        event.relation.data[self._charm.app]["alertmanager_config"] = json.dumps(
+            alertmanager_config
+        )
+
+    def config(self) -> dict:
+        """Returns Alertmanager configuration sent inside the relation data bag.
+
+        If the `alertmanager-remote-configuration` relation exists, takes the Alertmanager
+        configuration provided in the relation data bag and returns it in a form of a dictionary
+        if configuration passes the validation against the Alertmanager config schema.
+        If configuration fails the validation, error is logged and config is rejected (empty config
+        is returned).
+
+        Returns:
+            dict: Alertmanager configuration dictionary
+        """
+        config = {}
+        remote_configuration_relation = self._charm.model.get_relation(self._relation_name)
+        if remote_configuration_relation and remote_configuration_relation.app:
+            try:
+                raw_config = remote_configuration_relation.data[remote_configuration_relation.app][
+                    "alertmanager_config"
+                ]
+                if self._config_is_valid(json.loads(raw_config)):
+                    config = raw_config
+                else:
+                    logger.error("Config validation error.")
```

Comment:
I don't think we need a custom event here. At least I don't have a good idea of how we could leverage another event.
I agree that returning `{}` may not be the best idea tho. How about I just return `None` instead of replacing one invalid config with another?
In any case, in the charm, handling will still be the same:
```
remote_config, _ = self.remote_configuration_provider.config()
...
if remote_config:
    return yaml.safe_load(remote_config)
```
This treats `{}` and `None` equally. 
As for communicating a failure, `alertmanager-k8s` will not do it. In case there's no valid configuration coming from charm config or remote configuration relation, it falls back to the default config.

Reply:
This whole validation has been removed along with removing the JSON schema. Also method signature has been changed to return `Optional[dict]`. 
I guess that solves the problem.